Mathlib/Analysis/Distribution/SchwartzSpace.lean

```diff
@@ -1215,4 +1215,144 @@ def toZeroAtInftyCLM : 𝓢(E, F) →L[𝕜] C₀(E, F) :=
+  -- using `f₁ = (1 + ‖x‖) ^ (-k)` and `f₂ = (1 + ‖x‖) ^ k • f x`.
+  rcases hμ.exists_eLpNorm_lt_top p with ⟨k, hk⟩
+  refine ⟨k, (eLpNorm (fun x ↦ (1 + ‖x‖) ^ (-k : ℝ)) p μ).toNNReal * 2 ^ k, fun f ↦ ?_⟩
+  have h_one_add (x : E) : 0 < 1 + ‖x‖ := lt_add_of_pos_of_le zero_lt_one (norm_nonneg x)
+  refine le_trans ?_ ?_
+    (b := eLpNorm (fun x ↦ (1 + ‖x‖) ^ (-k : ℝ)) p μ * eLpNorm (fun x ↦ (1 + ‖x‖) ^ k • f x) ⊤ μ)
+  · refine le_of_eq_of_le ?_ (eLpNorm_smul_le_eLpNorm_mul_eLpNorm_top p _ ?_)
+    · congr
+      funext x
+      simp [Real.rpow_neg (h_one_add x).le, (h_one_add x).ne']
+    · refine Continuous.aestronglyMeasurable ?_
+      exact .rpow_const (.add continuous_const continuous_norm) fun x ↦ .inl (h_one_add x).ne'
+  · rw [ENNReal.coe_mul, mul_assoc]
+    refine mul_le_mul' ?_ ?_
+    · rw [ENNReal.coe_toNNReal hk.ne]
+    · rw [eLpNorm_exponent_top, ENNReal.ofReal, ← ENNReal.coe_mul]
+      refine eLpNormEssSup_le_of_ae_nnnorm_bound (ae_of_all μ fun x ↦ ?_)
+      rw [← norm_toNNReal, Real.toNNReal_le_iff_le_coe]
+      simpa [norm_iteratedFDeriv_zero, norm_smul, abs_of_nonneg (h_one_add x).le] using
+        one_add_le_sup_seminorm_apply (m := (k, 0)) (le_refl k) (le_refl 0) f x
+
+/-- The `L^p` norm of a Schwartz function is finite. -/
+theorem eLpNorm_lt_top (f : 𝓢(E, F)) (p : ℝ≥0∞) (μ : Measure E := by volume_tac)
+    [hμ : μ.HasTemperateGrowth] : eLpNorm f p μ < ⊤ := by
+  rcases eLpNorm_le_seminorm ℝ F p μ with ⟨k, C, hC⟩
+  exact lt_of_le_of_lt (hC f) (ENNReal.mul_lt_top ENNReal.coe_lt_top ENNReal.ofReal_lt_top)
+
+variable [SecondCountableTopologyEither E F]
+
+/-- Schwartz functions are in `L^∞`; does not require `hμ.HasTemperateGrowth`. -/
+theorem memℒp_top (f : 𝓢(E, F)) (μ : Measure E := by volume_tac) : Memℒp f ⊤ μ := by
+  rcases f.decay 0 0 with ⟨C, _, hC⟩
+  refine memℒp_top_of_bound f.continuous.aestronglyMeasurable C (ae_of_all μ fun x ↦ ?_)
+  simpa using hC x
+
+/-- Schwartz functions are in `L^p` for any `p`. -/
+theorem memℒp (f : 𝓢(E, F)) (p : ℝ≥0∞) (μ : Measure E := by volume_tac)
+    [hμ : μ.HasTemperateGrowth] : Memℒp f p μ :=
+  ⟨f.continuous.aestronglyMeasurable, f.eLpNorm_lt_top p μ⟩
+
+/-- Schwartz functions are in `L^p` for any `p`. -/
+def toLp (f : 𝓢(E, F))  (p : ℝ≥0∞) (μ : Measure E := by volume_tac) [hμ : μ.HasTemperateGrowth] :
+    Lp F p μ := (f.memℒp p μ).toLp
+
+theorem coeFn_toLp {f : 𝓢(E, F)} {p : ℝ≥0∞} {μ : Measure E} [hμ : μ.HasTemperateGrowth] :
+    f.toLp p μ =ᵐ[μ] f := (memℒp f p μ).coeFn_toLp
+
+theorem norm_toLp {f : 𝓢(E, F)} {p : ℝ≥0∞} {μ : Measure E} [hμ : μ.HasTemperateGrowth] :
+    ‖f.toLp p μ‖ = ENNReal.toReal (eLpNorm f p μ) := by
+  rw [Lp.norm_def, eLpNorm_congr_ae coeFn_toLp]
+
+variable (F) in
+/-- The map from Schwartz to `L^p` is injective when the measure is positive on non-empty sets. -/
+def embedding_toLp (p : ℝ≥0∞) (μ : Measure E := by volume_tac) [hμ : μ.HasTemperateGrowth]
```

Comment:
Instead of writing out a new (bundled) definition, it is often more convenient to state this as the fact that `toLp` is an injective function. In practice, the type of embeddings is not used a lot throughout mathlib for this reason.